robotics-88/vehicle-launch
# Cmu flight tests

Enabled switching the SLAM algorithm with a launch arg. There are now 3 options: fast_lio (our old default), fast_lio_lc, and faster_lio, corresponding to slam types 0, 1, and 2, respectively. 

**Airsim**
To test in AirSim, add the two new repos to your workspace, then launch with:
```
roslaunch vehicle_launch decco.launch simulate:=true min_alt:=-1 max_alt:=10 slam_type:=1
```
Repeat with slam types 0,1,2. Confirm mapping runs. Types 0, 2 seem to work fine in AirSim, but type 1 has a periodic rotation of 90 degrees. If the map looks good, try sending a polygon (command line or through hello decco) to start exploration and verify path planning works.

**Decco**
To test on the Decco NOT in flight:
```
roslaunch vehicle_launch decco.launch enable_autonomy:=false enable_exploration:=true min_alt:=-1 max_alt:=10 slam_type:=1
```
I typically set my desktop to use the Decco as ROS master so I can verify the map is running with 
```
roslaunch vehicle_launch viz_host_machine.launch
```
With both running, verify (a) maps are being created, (b) map <>slam_map tf is published, and (c) paths are generated. For (c), verify this by publishing a goal on the command line:
```
rostopic pub /goal geometry_msgs/PoseStamped '{header: {stamp: now, frame_id: "slam_map"}, pose: {position: {x: 20.0, y: 0.0, z: 5.0}, orientation: {w: 1.0}}}'
```
In Rviz, a red line should be generated from the drone to the goal, if a path was found.
